wandb/apis/public.py

```diff
@@ -4066,15 +4066,18 @@ def __repr__(self):
         return f"<ArtifactCollection {self.name} ({self.type})>"
 
 
-class _DownloadedArtifactEntry(artifacts.ArtifactEntry):
+class _DownloadedArtifactEntry(artifacts.ArtifactManifestEntry):
     def __init__(
-        self, name: str, entry: "artifacts.ArtifactEntry", parent_artifact: "Artifact"
+        self,
+        name: str,
+        entry: "artifacts.ArtifactManifestEntry",
+        parent_artifact: "Artifact",
     ):
         self.name = name
         self.entry = entry
         self._parent_artifact = parent_artifact
 
-        # Have to copy over a bunch of variables to get this ArtifactEntry interface
+        # Have to copy over a bunch of variables to get this ArtifactManifestEntry interface
```

Comment:
I think this whole block could/should be replaced by

```python
        super().__init__(
            path = entry.path,
            ref = entry.ref,
            digest = entry.digest,
            birth_artifact_id = entry.birth_artifact_id,
            size = entry.size,
            extra = entry.extra,
            local_path = entry.local_path,
        )
```

Reply:
Good point, that's much better. I also removed `entry` as a member to avoid double-storing everything, since after we've copied its values we don't need to keep it.